commitlint.config.js

```diff
@@ -0,0 +1,71 @@
+module.exports = {
+    rules: {
+        // rules are defined as [ severity (2=error), "always"|"never", argument? ]. See
+        // https://commitlint.js.org/#/reference-rules?id=rules
+
+        // log-commit-info is used for debugging. It outputs the dict passed to a plugin.
+        // 'log-commit-info': [2, 'always'],
+        // enforce that commits with a breaking change indicator ("!") must also contain an explanation of the breaking change.
+        'breaking-change-indicator': [2, 'always'],
+        // enforce that only the types below are allowed
+        'type-enum': [2, 'always',
+            [
+                'feat',
+                'fix',
+                'dep',
+                'ci',
+                'refactor',
+                'doc',
+                'test',
+            ]
+        ],
+        // enforce that a type is mandatory
+        'type-empty': [2, 'never'],
+        // enforce the maximum subject length (without type and scope)
+        'subject-max-length': [2, 'always', 75],
+        // enforce the maximum scope length
+        'scope-max-length': [2, 'always', 25],
+    },
+    parserPreset: {
+        parserOpts: {
+            // add matching groups for `!` before and after scope. Before scope is invalid, after scope is correct
+            // and enforces the "breaking-change" header (see below).
+            headerPattern: /^(\w*)(?:(!?)\(([\w\$\.\-\* ]*)\))?(!?)\: (.*)$/,
+            headerCorrespondence: ["type", "wrong_breaking", "scope", "breaking", "subject"],
+        },
+    },
+    plugins: [
+        {
+            rules: {
+                // log-commit-info is used for debugging. It outputs the dict passed to a plugin.
+                // 'log-commit-info': (input) => {
+                //     console.log(input)
+                //     return [true, "see above"]
+                // },
+                'breaking-change-indicator': ({wrong_breaking, breaking, scope, body, footer}) => {
+                    if (wrong_breaking) {
+                        return [
+                            false, "The breaking change indicator `!` can only be used right before `:`, e.g. `feat(scope)!:`."
+                        ]
+                    }
```

Comment:
```suggestion
                'breaking-change-indicator': ({wrong_breaking, breaking, scope, subject_prefix, body, footer}) => {
                    if (wrong_breaking) {
                        return [
                            false, "The breaking change indicator `!` can only be used right before `:`, e.g. `feat(scope)!:`."
                        ]
                    }
                    if (!subject_prefix) {
                        return [
                            false, "A space before the commit subject is required."
                        ]
+                    }
```

Reply:
> The following suggestions fix this but in a slightly hacky way (also the rule name breaking-change-indicator doesn't fit anymore).

As mentioned on the other comment, it's too hacky for me. Also for the purpose of our initial goal, i.e. "don't allow misfit commits", this does not add anything.

If you wanted, you can propose a new checker that does more than `type-empty` from the default set and inspects the commit message in full. But I see this separate to the breaking change indicator, and a non-vital improvement.